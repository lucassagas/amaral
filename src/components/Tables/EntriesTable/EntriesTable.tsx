import { EntriesProps } from '@/services/hooks/useEntries'
import { Table, Tbody, Td, Text, Th, Thead, Tr } from '@chakra-ui/react'

type EntriesTableProps = {
  entries?: EntriesProps[]
}

export function EntriesTable({ entries }: EntriesTableProps) {
  return (
    <Table colorScheme="whiteAlpha" mb="4">
      <Thead>
        <Tr>
          <Th>Produto</Th>
          <Th>Qtd. anterior</Th>
          <Th>Entrada</Th>
          <Th>Estoque</Th>
          <Th>Respons√°vel</Th>
          <Th>Data</Th>
        </Tr>
      </Thead>
      <Tbody>
        {entries?.map((entry) => (
          <Tr key={entry.id}>
            <Td>
              <Text>{entry.product.name}</Text>
            </Td>
            <Td>
              <Text>{entry.oldQuantity}</Text>
            </Td>
            <Td>
              <Text>{entry.quantity}</Text>
            </Td>
            <Td>
              <Text>{entry.newQuantity}</Text>
            </Td>
            <Td>
              <Text>{entry.createdBy.name}</Text>
            </Td>
            <Td>
              <Text>{entry?.updatedAt}</Text>
            </Td>
          </Tr>
        ))}
      </Tbody>
    </Table>
  )
}
