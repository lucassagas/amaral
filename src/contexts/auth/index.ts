export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
