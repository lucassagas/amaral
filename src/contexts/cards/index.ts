export { CardContext, CardContextProvider } from './cards';

