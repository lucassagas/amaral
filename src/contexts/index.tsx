import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
