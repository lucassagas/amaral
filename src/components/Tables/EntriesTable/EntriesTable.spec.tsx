import { AppProvider } from '@/contexts/index'
import { render } from '@testing-library/react'
import { EntriesTable } from './EntriesTable'

const mockedEntries = [
  {
    createdBy: {
      name: 'Lucas SagÃ¡s',
    },
    id: '03090998-a61b-4039-a0bb-565a9d90744b',
    newQuantity: 250,
    oldQuantity: 150,
    product: {
      name: 'peixe',
    },
    quantity: 100,
    updatedAt: '04/04/2022',
  },
]

const renderComponent = () => {
  return render(
    <AppProvider>
      <EntriesTable entries={mockedEntries} />
    </AppProvider>
  )
}

describe('<EntriesTable />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
