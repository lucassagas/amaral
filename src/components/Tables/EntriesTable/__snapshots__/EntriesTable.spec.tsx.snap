// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntriesTable /> should render correctly 1`] = `
<div>
  <table
    class="chakra-table css-e4kp03"
    role="table"
  >
    <thead
      class="css-0"
    >
      <tr
        class="css-0"
        role="row"
      >
        <th
          class="css-1ednkar"
        >
          Produto
        </th>
        <th
          class="css-1ednkar"
        >
          Qtd. anterior
        </th>
        <th
          class="css-1ednkar"
        >
          Entrada
        </th>
        <th
          class="css-1ednkar"
        >
          Estoque
        </th>
        <th
          class="css-1ednkar"
        >
          Responsável
        </th>
        <th
          class="css-1ednkar"
        >
          Data
        </th>
      </tr>
    </thead>
    <tbody
      class="css-0"
    >
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-0"
          >
            peixe
          </p>
        </td>
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-0"
          >
            150
          </p>
        </td>
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-0"
          >
            100
          </p>
        </td>
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-0"
          >
            250
          </p>
        </td>
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-0"
          >
            Lucas Sagás
          </p>
        </td>
        <td
          class="css-1n48h1c"
          role="gridcell"
        >
          <p
            class="chakra-text css-0"
          >
            04/04/2022
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
